src/devices/device_models/common/duration_constraints.jl

```diff
@@ -32,8 +32,8 @@ function device_duration_retrospective(ps_m::CanonicalModel,
                     tsd = max(1.0, duration_data[2][ix].down - initial_duration_off[ix].value)
                 end
 
-                ps_m.constraints[name_up][name, t] = JuMP.@constraint(ps_m.JuMPmodel, sum([ps_m.variables[var_names[2]][name,i] for i in ((t - tst + 1) :t) if i > 0 ]) <= ps_m.variables[var_names[1]][name,t])
-                ps_m.constraints[name_down][name, t] = JuMP.@constraint(ps_m.JuMPmodel, sum([ps_m.variables[var_names[3]][name,i] for i in ((t - tsd + 1) :t) if i > 0]) <= (1 - ps_m.variables[var_names[1]][name,t]))
+                ps_m.constraints[name_up][name, t] = JuMP.@constraint(ps_m.JuMPmodel, sum([ps_m.variables[var_names[2]][name,i] for i in ((t - round(tst) + 1) :t) if i > 0 ]) <= ps_m.variables[var_names[1]][name,t])
```

Comment:
This wouldn't work for models with 5-min resolution. Before adding this constraints add a warning

Reply:
I'll definitely add a warning for now. 
But as long as time_range is defined as UnitRange, fractional requirements will not work for any model. If we continue using this format then the Min-up/down requirement (also initial_duration_on/initial_duration_off) should be in no. of periods the unit has to be online/offline (regardless if 1 period is 5min or 1hr)

Reply:
time_range is the number of steps in the look_ahead. We can change that name to make it more clear.